src/storage/src/storage_state/async_storage_worker.rs

```diff
@@ -199,73 +201,12 @@ impl<T: Timestamp + Lattice + Codec64 + Display + Sync> AsyncStorageWorker<T> {
                 match command {
                     AsyncStorageWorkerCommand::UpdateFrontiers(id, ingestion_description) => {
-                        // Here we update the as-of and upper(i.e resumption) frontiers of the
-                        // ingestion.
-                        //
-                        // A good enough value for the as-of is the `meet({e.since for e in
-                        // exports})` but this is not as tight as it could be because the since
-                        // might be held back for unrelated to the ingestion reasons (e.g a user
-                        // wanting to keep historical data). To make it tight we would need to find
-                        // the maximum frontier at which all inputs to the ingestion are readable
-                        // and start from there. We can find this by defining:
-                        //
-                        // max_readable(shard) = {(t - 1) for t in shard.upper}
-                        // advanced_max_readable(shard) = advance_by(max_readable(shard), shard.since)
-                        // as_of = meet({advanced_max_readable(e) for e in exports})
-                        //
-                        // We defer this optimization for when Materialize allows users to
-                        // arbitrarily hold back collections to perform historical queries and when
-                        // the storage command protocol is updated such that these calculations are
-                        // performed by the controller and not here.
                         let mut resume_uppers = BTreeMap::new();
 
-                        // TODO(petrosagg): The as_of of the ingestion should normally be based
-                        // on the since frontiers of its outputs. Even though the storage
-                        // controller makes sure to make downgrade decisions in an organized
-                        // and ordered fashion, it then proceeds to persist them in an
-                        // asynchronous and disorganized fashion to persist. The net effect is
-                        // that upon restart, or upon observing the persist state like this
-                        // function, one can see non-sensical results like the since of A be in
-                        // advance of B even when B depends on A! This can happen because the
-                        // downgrade of B gets reordered and lost. Here is our best attempt at
-                        // playing detective of what the controller meant to do by blindly
-                        // assuming that the since of the remap shard is a suitable since
-                        // frontier without consulting the since frontier of the outputs. One
-                        // day we will enforce order to chaos and this comment will be deleted.
-                        let remap_shard = ingestion_description
+                        let seen_remap_shard = ingestion_description
```

Comment:
Nit: The `seen_` prefix here confuses me as I can't guess what it means! It kind of indicates that this isn't necessarily the actual remap shard?

Reply:
I will find a better name, that variable existed before the PR, I just moved it around